src/memex/search/index.py

```diff
@@ -143,13 +154,16 @@ def _stream_all_annotations(self, chunksize=2000):
 
         updated = self.session.query(models.Annotation.updated). \
                 execution_options(stream_results=True). \
+                filter_by(deleted=False). \
                 order_by(models.Annotation.updated.desc()).all()
 
         count = len(updated)
         windows = [Window(start=updated[min(x+chunksize, count)-1].updated,
                           end=updated[x].updated)
                    for x in xrange(0, count, chunksize)]
-        basequery = self._eager_loaded_query().order_by(models.Annotation.updated.asc())
+        basequery = self._eager_loaded_query(). \
+            filter_by(deleted=False). \
```

Comment:
This works ok, but rather than duplicating the filter in three places it might be nice to isolate the base query somewhere...

```python
def _base_query(session):
    return session.query(models.Annotation).filter_by(deleted=False)
```

I think the only wrinkle with that approach is reproducing your first query. Luckily I think that should be pretty straightforward, using `sqlalchemy.orm.load_only`:

```python
updated = (_base_query(self.session)
           .options(load_only('updated'))
           .execution_options(...)
           ...)
```

etc.